base/CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+cmake_minimum_required(VERSION 3.2)
+
+project(base C CXX)
+
+add_compile_options(
+  "-Wall"
+  "-pedantic"
+  "-std=c++11"
+)
+
+include_directories(
+  ${PROJECT_SOURCE_DIR}
+)
+
+set (CMAKE_SOURCE_DIR "${PROJECT_SOURCE_DIR}/base")
+add_subdirectory(base_tests)
+
+set(
+    HEADERS
+    SRC_FIRST.hpp
+    array_adapters.hpp
+    assert.hpp
+    base.hpp
+    bits.hpp
+    buffer_vector.hpp
+    cache.hpp
+    cancellable.hpp
+    collection_cast.hpp
+    condition.hpp
+    const_helper.hpp
+    deferred_task.hpp
+    exception.hpp
+    gmtime.hpp
+    limited_priority_queue.hpp
+    logging.hpp
+    macros.hpp
+    math.hpp
+    matrix.hpp
+    mem_trie.hpp
+    mutex.hpp
+    newtype.hpp
+    object_tracker.hpp
+    observer_list.hpp
+    range_iterator.hpp
+    ref_counted.hpp
+    rolling_hash.hpp
+    scope_guard.hpp
+    set_operations.hpp
+    shared_buffer_manager.hpp
+    src_point.hpp stats.hpp
+    std_serialization.hpp
+    stl_add.hpp
+    stl_helpers.hpp
+    stl_iterator.hpp
+    string_format.hpp
+    string_utils.hpp
+    strings_bundle.hpp
+    sunrise_sunset.hpp
+    swap.hpp
+    thread.hpp
+    thread_checker.hpp
+    thread_pool.hpp
+    threaded_container.hpp
+    threaded_list.hpp
+    threaded_priority_queue.hpp
+    timegm.hpp
+    timer.hpp
+    worker_thread.hpp
+    internal/message.hpp
+)
+
+set(
+    SOURCES
+    base.cpp
+    condition.cpp
+    condition_bada.cpp
+    condition_posix.cpp
+    condition_windows_native.cpp
+    deferred_task.cpp
+    exception.cpp
+    gmtime.cpp
+    logging.cpp
+    lower_case.cpp
+    normalize_unicode.cpp
+    object_tracker.cpp
+    shared_buffer_manager.cpp
+    src_point.cpp
+    string_format.cpp
+    string_utils.cpp
+    strings_bundle.cpp
+    sunrise_sunset.cpp
+    thread.cpp thread_checker.cpp
+    thread_pool.cpp
+    threaded_container.cpp
+    timegm.cpp
+    timer.cpp
+    internal/message.cpp
+)
+
+add_library(base ${HEADERS} ${SOURCES})
```

Comment:
Please, write a wrapper around add_library, that will link with threads and set target properties.